src/sentry/issues/escalating.py

```diff
@@ -435,10 +435,99 @@ def get_group_hourly_count(group: Group) -> int:
         hourly_count = int(
             raw_snql_query(request, referrer=IS_ESCALATING_REFERRER)["data"][0]["count()"]
         )
+        _get_group_hourly_count_with_metrics(group, now, current_hour, hourly_count)
         cache.set(key, hourly_count, GROUP_HOURLY_COUNT_TTL)
     return int(hourly_count)
 
 
+def _get_group_hourly_count_with_metrics(
+    group: Group, now: datetime, current_hour: datetime, hourly_count: int
+):
+    """
+    Return the number of events a group has had today in the last hour
+
+    Checks if the returned results are equivalent to `hourly_count`.
+    If not equivalent, it will generate a log.
+    """
+    organization = Organization.objects.get(id=group.project.organization_id)
+
+    if group.issue_category == GroupCategory.ERROR and features.has(
+        "organizations:escalating-issues-v2", organization
+    ):
+        metrics_query = _query_metrics_for_group_hourly_count(group, now, current_hour)
+        metrics_series_results = get_series(
+            projects=[group.project],
+            metrics_query=metrics_query,
+            use_case_id=UseCaseID.ESCALATING_ISSUES,
+        )
+
+        groups: List[Dict[str, Any]] = metrics_series_results.get("groups", [])
+
+        # Get the first item from groups or use a default empty dict
+        first_group: Dict[str, Any] = next(iter(groups), {})
+
+        # Get the "series" dict from the first group or default to an empty dict
+        series_dict: Dict[str, Any] = first_group.get("series", {})
+
+        # Finally, get the "event_ingested" list or default to an empty list
+        event_ingested: List[int] = series_dict.get("event_ingested", [])
+
+        # Extract the hourly count
+        metrics_hourly_count = next(iter(event_ingested), 0)
+
+        if metrics_hourly_count != hourly_count:
+            logger.info(
+                "_get_group_hourly_count_with_metrics results not the same as Errors dataset query.",
+                extra={
+                    "metrics_hourly_count": metrics_hourly_count,
+                    "hourly_count": hourly_count,
+                    "group_id": group.id,
+                    "now": now,
+                    "current_hour": current_hour,
+                },
+            )
+
+
+def _query_metrics_for_group_hourly_count(
+    group: Group, now: datetime, current_hour: datetime
+) -> MetricsQuery:
+    """
+    This function generates a query to fetch the current hour's events
+    for a group_id through the Generic Metrics Backend.
+
+    The Generic Metrics Backend only contains data for Errors.
+    """
+
+    if group.issue_category != GroupCategory.ERROR:
+        raise Exception("Invalid category.")
```

Comment:
Is raising an error going to disrupt any important processes? should we log and move on instead?